packages/members-api/lib/migrations/index.js

```diff
@@ -98,4 +101,120 @@ module.exports = class StripeMigrations {
+
+        if (!defaultStripeProduct) {
+            this._logging.info('Could not find Stripe Product');
+        }
+
+        for (const plan of plans) {
+            const price = await this.findPriceByPlan(plan);
+
+            if (!price) {
+                this._logging.info(`Could not find Stripe Price ${JSON.stringify(plan)}`);
+
+                try {
+                    this._logging.info(`Creating Stripe Price ${JSON.stringify(plan)}`);
+                    const price = await this._stripeAPIService.createPrice({
+                        currency: plan.currency,
+                        amount: plan.amount,
+                        nickname: plan.name,
+                        interval: plan.interval,
+                        active: true,
+                        type: 'recurring',
+                        product: defaultStripeProduct.get('stripe_product_id')
+                    });
+
+                    await this._StripePrice.add({
+                        stripe_price_id: price.id,
+                        stripe_product_id: defaultStripeProduct.get('stripe_product_id'),
+                        active: price.active,
+                        nickname: price.nickname,
+                        currency: price.currency,
+                        amount: price.unit_amount,
+                        type: 'recurring',
+                        interval: price.recurring.interval
+                    });
+                } catch (err) {
+                    this._logging.error({err, message: 'Adding price failed'});
+                }
+            }
+        }
+    }
+
+    async updatePortalPlansSetting(plans) {
+        this._logging.info('Migrating portal_plans setting from names to ids');
+        const portalPlansSetting = await this._Settings.findOne({key: 'portal_plans'});
+
+        let portalPlans;
+        try {
+            portalPlans = JSON.parse(portalPlansSetting.get('value'));
+        } catch (err) {
+            this._logging.error({
+                message: 'Could not parse portal_plans setting, skipping migration',
+                err
+            });
+            return;
+        }
+
+        const containsOldValues = !!portalPlans.find((plan) => {
+            return ['monthly', 'yearly'].includes(plan);
+        });
+
+        if (!containsOldValues) {
+            this._logging.info('Could not find names in portal_plans setting, skipping migration');
+            return;
+        }
+
+        const newPortalPlans = await portalPlans.reduce(async (newPortalPlansPromise, plan) => {
+            let newPlan = plan;
+            if (plan === 'monthly') {
+                const monthlyPlan = plans.find((plan) => {
+                    return plan.name === 'Monthly';
+                });
+                if (!monthlyPlan) {
+                    return newPortalPlansPromise;
+                }
+                const price = await this.findPriceByPlan(monthlyPlan);
+                newPlan = price.id;
```

Comment:
I think we need to guard against `price` not being found in DB, as there is a chance that due to failure in earlier migration we don't have the price and then this will fail. Ran across this while testing a fresh setup with new launch wizard.